libraries/AP_RangeFinder/AP_RangeFinder_NRA24.cpp

```diff
@@ -0,0 +1,100 @@
+/*
+ * Copyright (C) 2016  Intel Corporation. All rights reserved.
+ *
+ * This file is free software: you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License as published by the
+ * Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This file is distributed in the hope that it will be useful, but
+ * WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+ * See the GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License along
+ * with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include <AP_HAL/AP_HAL.h>
+#include <AP_SerialManager/AP_SerialManager.h>
+#include <ctype.h>
+#include "AP_RangeFinder_NRA24.h"
+
+//#define MAXSONAR_SERIAL_LV_BAUD_RATE 115200
+
+extern const AP_HAL::HAL& hal;
+
+/* 
+   The constructor also initialises the rangefinder. Note that this
+   constructor is not called until detect() returns true, so we
+   already know that we should setup the rangefinder
+*/
+AP_RangeFinder_NRA24::AP_RangeFinder_NRA24(RangeFinder &_ranger, uint8_t instance,
+                                                                 RangeFinder::RangeFinder_State &_state,
+                                                                 AP_SerialManager &serial_manager) :
+    AP_RangeFinder_Backend(_ranger, instance, _state, MAV_DISTANCE_SENSOR_ULTRASOUND)
+{
+    uart = serial_manager.find_serial(AP_SerialManager::SerialProtocol_Lidar, 0);
+    if (uart != nullptr) {
+        uart->begin(serial_manager.find_baudrate(AP_SerialManager::SerialProtocol_Lidar, 0));
+    }
+}
+
+/* 
+   detect if a MaxSonar rangefinder is connected. We'll detect by
+   trying to take a reading on Serial. If we get a result the sensor is
+   there.
+*/
+bool AP_RangeFinder_NRA24::detect(RangeFinder &_ranger, uint8_t instance, AP_SerialManager &serial_manager)
+{
+    return serial_manager.find_serial(AP_SerialManager::SerialProtocol_Lidar, 0) != nullptr;
+}
+
+// read - return last value measured by sensor
+bool AP_RangeFinder_NRA24::reading(uint16_t &cm)
+{
+    const unsigned char head[] = {0xAA, 0xAA, 0x0C, 0x07};
+
+    if (uart == nullptr) {
+        return false;
+    }
+
+    while (uart->available()){
```

Comment:
Probably not a good idea to process as much input as you can; if you vehicle is too busy working out how high it is it may not actually fly itself...